File: backend/apps/statistics/services/statistics_service.py
"""
Statistics Service.
Aggregates counts from multiple collections.
"""

from apps.common.database.mongo import mongo
from apps.common.core.collections import Collections


class StatisticsService:
    """Aggregates dashboard statistics from MongoDB."""

    @staticmethod
    def get_dashboard_stats():
        """Return summary counts for the dashboard."""
        users_collection = mongo.get_collection(Collections.USERS)
        departments_collection = mongo.get_collection(Collections.DEPARTMENTS)
        attendance_collection = mongo.get_collection(Collections.ATTENDANCE)
        leaves_collection = mongo.get_collection(Collections.LEAVES)

        total_employees = users_collection.count_documents({})
        total_departments = departments_collection.count_documents({})
        total_attendance = attendance_collection.count_documents({})
        pending_leaves = leaves_collection.count_documents({"status": "PENDING"})

        return {
            "total_employees": total_employees,
            "total_departments": total_departments,
            "total_attendance": total_attendance,
            "pending_leaves": pending_leaves,
        }

File: frontend/src/services/statistics.service.ts
/**
 * Statistics API service.
 */

import { http } from "./api";

export interface DashboardStatistics {
  total_employees: number;
  total_departments: number;
  total_attendance: number;
  pending_leaves: number;
}

export const statisticsService = {
  async getDashboardStats(): Promise<DashboardStatistics> {
    return http.get<DashboardStatistics>("/statistics/");
  },
};
